### PR TITLE
scheduler metrics should only observe successful ops

### LINKED ISSUE #19246: Scheduler latency metrics should separate error case from normal
Currently scheduler latency [metrics](https://github.com/kubernetes/kubernetes/blob/c4fdb7a2d06db250f577f06e41182a9f22cc8f7c/plugin/pkg/scheduler/metrics/metrics.go#L31) are observed regardless of failure. For example, binding latency is [recorded](https://github.com/kubernetes/kubernetes/blob/c4fdb7a2d06db250f577f06e41182a9f22cc8f7c/plugin/pkg/scheduler/scheduler.go#L152) like:
```Go
err := s.config.Binder.Bind(b)
		metrics.BindingLatency.Observe(metrics.SinceInMicroseconds(bindingStart))
		if err != nil {
			glog.V(1).Infof("Failed to bind pod: %+v", err)
			...
			return
		}
```

This could be a potential confusion if latency spikes happen because of failure. We should better handle it like separating error case from normal

### PR DESCRIPTION
**Pull request by hongchaodeng**
This resolves #19246 
We should record metrics for successful ops only without interference from failure cases. I'm not sure if we should separately record failure cases as well? At least for here?

@gmarek

### CODE REVIEW

File: plugin/pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -149,17 +147,19 @@ func (s *Scheduler) scheduleOne() {
 	s.config.Modeler.LockedAction(func() {
 		bindingStart := time.Now()
 		err := s.config.Binder.Bind(b)
-		metrics.BindingLatency.Observe(metrics.SinceInMicroseconds(bindingStart))
 		if err != nil {
 			glog.V(1).Infof("Failed to bind pod: %+v", err)
 			s.config.Recorder.Eventf(pod, api.EventTypeNormal, "FailedScheduling", "Binding rejected: %v", err)
 			s.config.Error(pod, err)
 			return
```
Comment (gmarek) [inline plugin/pkg/scheduler/scheduler.go]: I guess you should add a metric for errorrs in binding and record the latency here.

**Comment by hongchaodeng** [comment]
Added failed binding latency metrics @gmarek

**Comment by gmarek** [comment]
I'll be a pain, but my impression is that prometheus suggest using histograms instead of summaries: http://prometheus.io/docs/practices/histograms/.

I know we're not doing it right currently, but it's never too late to start:)

File: plugin/pkg/scheduler/metrics/metrics.go [hunk 2]
```diff
@@ -52,6 +52,14 @@ var (
 			MaxAge:    time.Hour,
 		},
 	)
+	FailedBindingLatency = prometheus.NewSummary(
```
Comment (xiang90) [inline plugin/pkg/scheduler/metrics/metrics.go]: I think a failedBinding counter is good enough. But a latency summary also wont hurt.

**Comment by hongchaodeng** [comment]
Changed all Summary to Histogram.

I saw Xiang's comment on failed binding latency metrics. Should we counter instead? @gmarek

**Comment by gmarek** [comment]
I think that we don't need failed binding on the scheduler side, at least for now. We do have `apiserver_request_count` metric in the API server, e.g
`12:42:19 	[code=200,resource=bindings,verb=POST,client=kube-scheduler/v1.2.0 (linux/amd64) kubernetes/2cd8bff] = 3823`
so long as we have single scheduler it's good enough. It wouldn't hurt to add it to be futureproof though, so I won't object adding those.

Also, can you please add new metrics to https://github.com/kubernetes/kubernetes/blob/master/pkg/metrics/scheduler_metrics.go#L25?

**Comment by hongchaodeng** [comment]
Removed failed binding metrics.
We can observe from logs/events that failure happened, which is an appropriate place to observe failure.

**Comment by gmarek** [comment]
I meant failed binding count, not failed binding latencies - sorry, for the confusion, working late just after holidays makes this to people (at least to me:). I think we can merge this as it is and add a metric later if it'll be proven useful.

Thanks.

**Pull request merged**